cmd/otelcol/main.go

```diff
@@ -239,63 +167,67 @@ func checkRuntimeParams() {
-// Sets '--config' to default config file path if '--config', SPLUNK_CONFIG and SPLUNK_CONFIG_YAML not specified.
-func checkConfig() {
-	configPathFlagExists, configPathFlag := getKeyValue(os.Args[1:], "--config")
+func checkInputConfigs(inputFlags flags) {
 	configPathVar := os.Getenv(configEnvVarName)
 	configYaml := os.Getenv(configYamlEnvVarName)
 
-	if configPathFlagExists && configPathFlag == "" {
-		log.Fatal("Command line flag --config specified but empty")
+	for _, filePath := range inputFlags.getConfigFlags() {
+		if _, err := os.Stat(filePath); err != nil {
+			log.Fatalf("Unable to find the configuration file (%s) ensure flag '--config' is set properly", filePath)
+		}
 	}
 
-	if configPathFlag != "" {
-		if _, err := os.Stat(configPathFlag); err != nil {
-			log.Fatalf("Unable to find the configuration file (%s) ensure flag '--config' is set properly", configPathFlag)
-		}
+	if configPathVar != "" && !inputFlags.configFlags.contains(configPathVar) {
+		log.Printf("Both environment variable %v and flag '--config' were specified. Using the flag values and ignoring the environment variable value %s in this session", configEnvVarName, configPathVar)
+	}
 
-		if configPathVar != "" && configPathVar != configPathFlag {
-			log.Printf("Both environment variable %v and flag '--config' were specified. Using the flag value %s and ignoring the environment variable value %s in this session", configEnvVarName, configPathFlag, configPathVar)
-		}
+	if configYaml != "" {
+		log.Printf("Both environment variable %s and flag '--config' were specified. Using the flag values and ignoring the environment variable in this session", configYamlEnvVarName)
+	}
 
-		if configYaml != "" {
-			log.Printf("Both environment variable %s and flag '--config' were specified. Using the flag value %s and ignoring the environment variable in this session", configYamlEnvVarName, configPathFlag)
-		}
+	checkRequiredEnvVars(inputFlags.getConfigFlags())
+}
 
-		checkRequiredEnvVars(configPathFlag)
+func checkConfigPathEnvVar(inputFlags flags) {
+	configPathVar := os.Getenv(configEnvVarName)
+	configYaml := os.Getenv(configYamlEnvVarName)
 
-		log.Printf("Set config to %v", configPathFlag)
-		return
+	if _, err := os.Stat(configPathVar); err != nil {
+		log.Fatalf("Unable to find the configuration file (%s) ensure %s environment variable is set properly", configPathVar, configEnvVarName)
 	}
 
-	if configPathVar != "" {
-		if _, err := os.Stat(configPathVar); err != nil {
-			log.Fatalf("Unable to find the configuration file (%s) ensure %s environment variable is set properly", configPathVar, configEnvVarName)
-		}
+	if configYaml != "" {
+		log.Printf("Both %s and %s were specified. Using %s environment variable value %s for this session", configEnvVarName, configYamlEnvVarName, configEnvVarName, configPathVar)
+	}
 
-		os.Args = append(os.Args, "--config="+configPathVar)
+	if !inputFlags.configFlags.contains(configPathVar) {
+		inputFlags.configFlags.Set(configPathVar)
```

Comment:
GoLand complains about these as unhandled error (even though the implementation never returns an error). Can we ignore explicitly?

```
_ = inputFlags.configFlags.Set(configPathVar)
```

Reply:
Fixed, thanks!